### PR TITLE
Add forwardproxy to directives' list

### PR DESCRIPTION
**Pull request by sergeyfrolov**
### 1. What does this change do, exactly?
Adds forwardproxy to directives' list

### 2. Please link to the relevant issues.
N/A

### 3. Which documentation changes (if any) need to be made because of this PR?
ASAIK, the documentation for plugins is handled at registration.

### 4. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I am willing to help maintain this change if there are issues with it later

### CODE REVIEW

File: caddyhttp/httpserver/plugin.go [hunk 1]
```diff
@@ -449,6 +449,7 @@ var directives = []string{
 
 	// directives that add listener middleware to the stack
 	"proxyprotocol", // github.com/mastercactapus/caddy-proxyprotocol
+	"forwardproxy",  // github.com/caddyserver/forwardproxy
```
Comment (mholt) [inline caddyhttp/httpserver/plugin.go]: This group is actually for plugins which implement [httpserver.ListenerMiddleware](https://godoc.org/github.com/mholt/caddy/caddyhttp/httpserver#ListenerMiddleware) which this does not (although I can see why you put it here) -- no big deal, just move it down to between `// directives that add middleware to the stack` and `"locale",` - so, still in the same place, but right now the grouping is just a little confusing. I should probably improve the wording of the comments too.

**Comment by sergeyfrolov** [comment]
Oh, sure, my bad.
I moved it, not sure if new place is the best, but we can reorder later.

**Comment by elvizlai** [comment]
When will this plugin work? I saw doc on caddy website, but no choice for built.

**Comment by mholt** [comment]
As soon as @sergeyfrolov releases it. :) The website is ready for it.

**Comment by sergeyfrolov** [comment]
@elvizlai @mholt thanks for your interest, `forwardproxy` is released!
It works for me, but if you encounter any issues, please file them at [forwardproxy repo](https://github.com/caddyserver/forwardproxy).

**Pull request merged**